src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -291,21 +338,33 @@ private void InitializeConstructorParameters(Dictionary<string, JsonPropertyInfo
             PropertyCache = propertyCache;
         }
 
-        private static bool PropertyIsOverridenAndIgnored(PropertyInfo currentProperty, Dictionary<string, PropertyInfo>? ignoredProperties)
+        private static bool PropertyIsOverridenAndIgnored(MemberInfo currentMember, Dictionary<string, MemberInfo>? ignoredMembers)
         {
-            if (ignoredProperties == null || !ignoredProperties.TryGetValue(currentProperty.Name, out PropertyInfo? ignoredProperty))
+            if (ignoredMembers == null || !ignoredMembers.TryGetValue(currentMember.Name, out MemberInfo? ignoredProperty))
             {
                 return false;
             }
 
-            return currentProperty.PropertyType == ignoredProperty.PropertyType &&
-                PropertyIsVirtual(currentProperty) &&
-                PropertyIsVirtual(ignoredProperty);
+            Debug.Assert(currentMember is PropertyInfo || currentMember is FieldInfo);
+            PropertyInfo? currentPropertyInfo = currentMember as PropertyInfo;
+            Type currentMemberType = currentPropertyInfo == null
+                ? Unsafe.As<FieldInfo>(currentMember).FieldType
+                : currentPropertyInfo.PropertyType;
+
+            Debug.Assert(ignoredProperty is PropertyInfo || ignoredProperty is FieldInfo);
+            PropertyInfo? ignoredPropertyInfo = ignoredProperty as PropertyInfo;
+            Type ignoredPropertyType = ignoredPropertyInfo == null
+                ? Unsafe.As<FieldInfo>(ignoredProperty).FieldType
+                : ignoredPropertyInfo.PropertyType;
```

Comment:
Copy/paste error?

Reply:
What do you mean?

Reply:
It appeared that this chunk of code was repeated. Either my eyes tricked me, or it was resolved.

Reply:
No, that code hasn't changed, but it really looks like the previous lines (:

Reply:
AH, got it.